.github/workflows/ci.yml

```diff
@@ -0,0 +1,106 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+
+  merge_group:
+
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  prefetch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.1
+
+      - name: Update pip
+        run: |
+          pip install --upgrade pip
+          pip --version
+
+      - name: Install poetry
+        run: |
+          pip install --upgrade poetry
```

Comment:
nit: I think we should pin the poetry version we're using here just to stay consistent between dev and CI https://python-poetry.org/docs/#ci-recommendations

Reply:
Good idea, done.